server/src/main/java/org/jetbrains/bsp/bazel/server/diagnostics/CompilerDiagnosticParser.kt

```diff
@@ -23,12 +23,17 @@ object CompilerDiagnosticParser : Parser {
           ?.let { match ->
             val path = match.groupValues[1]
             val line = match.groupValues[2].toInt()
-            val column = match.groupValues[3].toIntOrNull() ?: 1
+            val column = match.groupValues[3].toIntOrNull() ?: tryFindColumnNumber(output) ?: 1
             val level = if (match.groupValues[4] == "warning") Level.Warning else Level.Error
-            val message = constructMessage(match.value, output)
+            val message = constructMessage(match.groupValues[5], output)
             Diagnostic(Position(line, column), message, level, path, targetLabel)
           }
 
+  private fun tryFindColumnNumber(output: Output): Int? {
+    val line = output.peek(limit = 20).find { IssuePositionMarker.matches(it) }
```

Comment:
Maybe it would be cleaner to take `message` and look for `^` there? In theory you can peek something else here and you are skipping the logic of constructMessage which can change for example.